server.js

```diff
@@ -1,30 +1,80 @@
 import express from "express";
 import cors from "cors";
+import booksData from "./data/books.json";
+import listEndpoints from "express-list-endpoints";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
+// Defines the port the app will run on. Defaults to 8080, but can be overridden when starting the server. 
+// Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
+// Middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// Endpoint to fetch all books
+app.get("/books", (req, res) => {
+  res.json(booksData);
+});
+
+// Function to filter books by their language
+const filterBooksByLanguage = (req, res) => {
+  const { language } = req.query;
+
+  // Return an error if language query parameter is missing
+  if (!language) {
+    return res.status(400).json({
+      error: "Language parameter is required.",
+      example_usage: "Try this: /books/language?language=spa",
+      query_parameter_required: "language",
+      description: "Filter books by language using the language_code (e.g., 'spa', 'en', 'fr')."
+    });
+  }
+
+  // Filter books matching the specified language
+  const filteredBooks = booksData.filter(
+    book => book.language_code.toLowerCase() === language.toLowerCase()
+  );
+
+  // Return the filtered books if matches are found, or show error
+  if (filteredBooks.length > 0) {
+    res.json(filteredBooks);
+  } else {
+    res.status(404).json({
+      error: `No books found for this language: ${language}`,
+      example_usage: "Try this: /books/language?language=spa"
+    });
+  }
+};
+
+// Endpoint to fetch books filtered by language
+app.get("/books/language", filterBooksByLanguage);
```

Comment:
It would be more RESTful if this was a query param in the /books endpoint instead of its own endpoint, e.g. `/books?language=english`